### DISCUSSION TITLE
GRDB.swift on Windows

### DISCUSSION THREAD

**Discussion by arasan01** [groue/GRDB.swift#1498] [Show and tell]
Tweaks were made to make GRDB.swift run on Windows to explore the possibility of using it on Windows.

https://github.com/arasan01/GRDB.swift

The ability to use GRDB.swift to handle SQLite conveniently in the Windows world is a huge advantage. So far we have had to exclude a lot of testing. We also think we need to prepare something suitable for Windows, such as smoke tests, but using GRDB.swift together with the Windows App seems realistic. Why not try it out and share your thoughts?

![GG8qjk3bMAAcAaU](https://github.com/groue/GRDB.swift/assets/15707221/7129c06b-655f-4993-86f2-49530d01b83d)

This Fork is not considering submitting a PR to groue/GRDB.swift at this time. The reason is to respect the opinion of #1359 Windows support would still be a huge burden for groue/GRDB.swift to manage, and this is not merge-worthy.

**Comment by groue** [comment]
Hello @arasan01,

Welcome, and thank you very much for your bringing GRDB to Windows 🤩!

Thank you also for considering #1359, this is appreciated. Some pull requests are merged nevertheless, such as #1401. As you have understood, I don't promise that the support for untested platforms won't be broken at some point. But raising the awareness is still an advantage, because not making promises doesn't mean deliberately sabotaging the work of others!

**Reply by groue** [reply]
Actually this is big. Don't hesitate cross-posting your announcement online or in the GRDB forums, at https://forums.swift.org/c/related-projects/grdb/36, if you want some publicity and more feedback.

cc @compnerd

**Comment by arasan01** [comment]
Progress report. We have created one that actually runs as a Windows application. It appears to be working very well.

https://github.com/groue/GRDB.swift/assets/15707221/6c5d0ddf-f1d5-4d9c-b513-d7a64df3b8d8

Here is the sample code. 

https://github.com/arasan01/swift-windows-application

**Reply by arasan01** [reply]
@groue 
I felt this when I actually created a simple GUI application. I have a feeling that this will be quite useful. Therefore, I will prepare to release PR as an experimental feature. At this point, I am not familiar with the CI environment in Windows, but do we need a complete and exhaustive test code, etc.? I am not sure at what level of completeness I should submit the PR. 
I would like to hear @groue opinion.

**Reply by groue** [reply]
Hello @arasan01,

I'm currently far away from any computer: please wait a few days for a detailed feedback. I'm very interested in your contributions, so be sure I'll do my best to help you!

**Reply by groue** [reply]
> At this point, I am not familiar with the CI environment in Windows, but do we need a complete and exhaustive test code, etc.? I am not sure at what level of completeness I should submit the PR.

Oh, I understand your question.

Test as much as you want. I will not judge the pull request on the quantity of Windows tests that run and pass - zero is OK. I will only judge the pull request on the quantity of tests that break. Tests that break are a hindrance for everyone: pull requests are red, the main branch is red, etc. That's what I want to avoid.

Also, please remember that all I have is the CI output - I can't run Windows on a local machine, I don't have access to a debugger, etc.

**If you commit changes to `.github/workflows/CI.yml`, you will become responsible for them.**

Precisely, this means that I will take care of preserving them, until they the CI starts to fail on Windows tests. At that point:

I will try to solve the problem on my own. If I can't, I will ask for your help. If you don't answer with an effective solution in a reasonable delay, I will remove Windows tests from `.github/workflows/CI.yml`. Windows tests will be restored as soon as you come back with a solution.

If I have to frequently run this dance (politely but impatiently waiting for your feedback, eventually removing Windows tests, and restoring them after a while), I will question your commitment and might want to stop and terminate the collaboration. There is no hard-defined rule here. I will be respectful of your time just as I am of all GRDB users, and I'm aware we're all doing this on our free time. I expect the same from you.

Would you be OK with that? Does this sound like we could get along?
